### PR TITLE
554-🎨⭐- Add Language choice for First Screen

### PR DESCRIPTION
**Pull request by Esquimor**
Link to this issue: [#554](https://github.com/archethic-foundation/archethic-wallet/issues/554)

What I have done:

- Refacto: intro_welcom.dart
- Add a button to change language
- Create the component "BannerConnectivityNotAvailable" to use the same component everywhere.

@redDwarf03 you can review

### CODE REVIEW

File: lib/ui/widgets/components/banner_connectivity_not_available.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/// SPDX-License-Identifier: AGPL-3.0-or-later
+import 'package:aewallet/ui/util/main_appBar_icon_network_warning.dart';
+// Package imports:
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class BannerConnectivityNotAvailable extends ConsumerWidget {
```
Comment (redDwarf03) [inline lib/ui/widgets/components/banner_connectivity_not_available.dart]: Good proposal. Just perhaps instead this new widget, add in MainAppBarIconNetworkWarning a new optional  param "Alignment alignment". By default, alignment could be center but in intro pages, you can pass ``Alignment.topRight``.

Comment (redDwarf03) [reply lib/ui/widgets/components/banner_connectivity_not_available.dart]: And rename ``MainAppBarIconNetworkWarning``to ``IconNetworkWarning`` and move to your folder proposal ``lib/ui/widgets/components``

File: lib/ui/views/intro/intro_welcome.dart [hunk 2]
```diff
@@ -258,3 +81,328 @@ class _IntroWelcomeState extends ConsumerState<IntroWelcome> {
...
+              height: 200,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _WelcomTextFirst extends ConsumerWidget {
```
Comment (redDwarf03) [inline lib/ui/views/intro/intro_welcome.dart]: please, "WelcomeTextFirst"

File: lib/ui/views/intro/intro_welcome.dart [hunk 3]
```diff
@@ -258,3 +81,328 @@ class _IntroWelcomeState extends ConsumerState<IntroWelcome> {
...
+      child: AutoSizeText(
+        localizations.welcomeText,
+        maxLines: 3,
+        style: theme.textStyleSize16W400Primary,
+      ),
+    );
+  }
+}
+
+class _WelcomTextSecond extends ConsumerWidget {
```
Comment (redDwarf03) [inline lib/ui/views/intro/intro_welcome.dart]: please, "WelcomeTextSecond"

File: lib/ui/views/intro/intro_welcome.dart [hunk 4]
```diff
@@ -258,3 +81,328 @@ class _IntroWelcomeState extends ConsumerState<IntroWelcome> {
...
+  const _Language();
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return Align(
+      alignment: Alignment.topLeft,
+      child: IconButton(
+        icon: const Icon(
+          UiIcons.language,
+          color: Colors.blue,
```
Comment (redDwarf03) [inline lib/ui/views/intro/intro_welcome.dart]: This color is not referenced in the app. Perhaps use theme.iconDrawer

File: lib/ui/views/intro/intro_welcome.dart [hunk 5]
```diff
@@ -258,3 +81,328 @@ class _IntroWelcomeState extends ConsumerState<IntroWelcome> {
...
+        ),
+        _ButtonImportWallet(
+          checkedValue: checkedValue,
+        ),
+      ],
+    );
+  }
+}
+
+class _CGU extends ConsumerWidget {
```
Comment (redDwarf03) [inline lib/ui/views/intro/intro_welcome.dart:273]: Continue your idea of structuration: separate CGU and version info.
So don't use isConnectivityAvailable in param but directly a condition to display or not the widget ``_CGU``

File: lib/ui/views/intro/intro_welcome.dart [hunk 6]
```diff
@@ -258,3 +81,328 @@ class _IntroWelcomeState extends ConsumerState<IntroWelcome> {
...
+            ),
+          );
+          Navigator.of(context).pushNamed('/intro_import');
+        }
+      },
+    );
+  }
+}
+
+class _ButtonAction extends ConsumerWidget {
```
Comment (redDwarf03) [inline lib/ui/views/intro/intro_welcome.dart:421]: Could you change the name of ``outline``param with ``enabled`` please
And the button import should have ``Dimens.buttonBottomDimens`` and create button ``  Dimens.buttonTopDimens``
I prefer you let buttons like before (perhaps in this case, you are going too far in structuration ;) )

**Review by redDwarf03 — CHANGES_REQUESTED** [review]

**Comment by Esquimor** [comment]
I have resolve all the changes.

Also, I have put the language icon in an AppBar. Can you tell me if it suit ? @redDwarf03

**Comment by redDwarf03** [comment]
i changed the UI:
- i add the warning connectivity icon in the appbar 
- i fix the background displayed

Thx for contribution @Esquimor 
i merge the PR

**Review by redDwarf03 — APPROVED** [review]

**Pull request merged**